src/main/java/org/mitre/hapifhir/MedMorphToCIBMTR.java

```diff
@@ -155,10 +181,31 @@ protected void postBundle(String authToken, String ccn, List<BundleEntryComponen
     }
   }
 
-  protected JSONArray getObservationEntries(String ccn, List<BundleEntryComponent> observationEntries, String resourceId) {
+  protected JSONArray getObservationEntries(CloseableHttpClient httpClient, String authToken, String ccn, List<BundleEntryComponent> observationEntries, String resourceId) throws Exception {
```

Comment:
One performance thought here is that if we know the Patient resource is new then we don't need to check to see if the Observations already exist, we know they don't. I'm not sure how many calls we are going to be sending through, but if we can safely reduce that number then I think we should. I'd suggest a new boolean flag for `isPatientNew` or something like that, with a default of true, then if the check on lines 103-120 finds anything then set it to true. Then pass that as a new param into this method which allows you to skip the dup check.

Reply:
Another thought on performance, I wonder if it's preferable to query for all Observations for the given Patient, or to query individually by ID. I think the real answer will depend heavily on the real CIBMTR implementation, so no point in us trying to guess. Let's leave the current approach of query-by-id and that can be something that evolves during testing.

Reply:
I added the `isPatientNew` flag. As for your other comment, it might make more sense to make one query to get all the Observations for the given Patient and then checking the ID locally but I'll leave it as is for now until we can test with the actual implementation.